### PR TITLE
[FIX] mail: add open chat when clicking on author im status icon

### PR DESCRIPTION
**Pull request by seb-odoo**
task-2329538

### CODE REVIEW

File: addons/mail/static/src/components/message/message.xml [hunk 1]
```diff
@@ -27,6 +27,7 @@
                                         'o-message-selected': props.isSelected,
                                         'o_Message_partnerImStatusIcon-mobile': env.messaging.device.isMobile,
                                     }"
+                                    hasOpenChat="hasAuthorOpenChat"
```
Comment (alexkuhn) [inline addons/mail/static/src/components/message/message.xml:30]: `hasAuthorOpenChat` code looks wrong for disabling open chat on author that are sole correspondent of current chat: it should use `this.threadView.thread`, not `this.message.originThread`...

In theory message could originate from chat and a channel may follow this chat, so in context of channel we should be able to open chat.

**Review by alexkuhn — COMMENTED** [review]
LGTM, just see something's off with `hasAuthorOpenChat`, so I would fix it in this PR at the same time.

File: addons/mail/static/src/components/message/message.xml (hunk 1 above)
Comment (seb-odoo) [reply addons/mail/static/src/components/message/message.xml:30]: Test will require a thread view(er) to work if I make that change, but I guess it is indeed more correct.

Comment (alexkuhn) [reply addons/mail/static/src/components/message/message.xml:30]: If you really want test (which is better), could wait for message_fetch PR for having thread viewer.

Comment (seb-odoo) [reply addons/mail/static/src/components/message/message.xml:30]: I mean I added a test here already, but it's a message test so I'll have to give it the proper manual thread view id. But it's ok I'll just do it like other tests in that file.

Comment (seb-odoo) [reply addons/mail/static/src/components/message/message.xml:30]: done

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**